client/src/components/DisplayTable/DisplayTable.stories.tsx

```diff
@@ -37,24 +46,25 @@ Basic.args = {
   column_configs: common_column_configs,
 };
 
-const country_lookup = {
+const country_lookup: Record<string, string> = {
   Italy: "Italy",
   Korea: "Korea",
   Canada: "Canada",
   United_States: "United States",
   United_Kingdom: "United Kingdom",
 };
+
 const table_with_URL_column_configs = {
   ...common_column_configs,
   label: {
     ...common_column_configs.label,
-    formatter: (id) => (
+    formatter: (id: CellValue) => (
       <a
         href={`https://en.wikipedia.org/wiki/List_of_pizza_varieties_by_country#${id}`}
         target="_blank"
         rel="noopener noreferrer"
       >
-        {country_lookup[id]}
+        {typeof id === "string" ? country_lookup[id] : <></>}
```

Comment:
Could I just cast id to string here?

Reply:
TS is supposed to help us write safer run time code. Painful type system situations could be a good hint that more run time checks should be added (ideally ones that type narrow). Best I could come up with that's both TS-correct and runtime safe is something like this:
```
const country_lookup = {
  Italy: "Italy",
  Korea: "Korea",
  Canada: "Canada",
  United_States: "United States",
  United_Kingdom: "United Kingdom",
};

const table_with_URL_column_configs = {
  ...common_column_configs,
  label: {
    ...common_column_configs.label,
    formatter: (label: CellValue) => {
      const country_name = _.get(country_lookup, _.toString(label));

      if (typeof country_name !== "undefined") {
        return (
          <a
            href={`https://en.wikipedia.org/wiki/List_of_pizza_varieties_by_country#${label}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            {country_name}
          </a>
        );
      } else {
        throw new Error("Something to explain what this situation might be");
      }
    },
  },
};
```

In theory we could "know" `label` will always be a string (and use an `as` to cast its type) but that'll still leave TS complaining that `string` is too broad for `country_lookup` (note I removed the `Record<string,string>` type) which is not wrong. So either way this could use an extra layer of runtime checking for whether the `label` value was actually in `country_lookup`.

Reply:
... it's still pretty painful though, don't want to be wrestling with the over-broad `CellValue` type on every (TS) use of `DisplayTable`. I'll think about if it can be made easier to use with generics or something.